### PR TITLE
[8.0][IMP] stock move write in block

### PR DESCRIPTION
**Pull request by sergiocorato**
Description of the issue/feature this PR addresses: write is slow as it is called record by record

Current behavior before PR: write is slow as it is called record by record

Desired behavior after PR is merged: write is faster as it is called one time on all records

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by gurneyalex — APPROVED** [review]
Just curious: did you measure an actual speed improvement?

**Comment by sergiocorato** [comment]
> Just curious: did you measure an actual speed improvement?

With 114 moves it need 287.885869026s with this PR vs. 496.347542048s without, 42% faster.

**Review by mileo — APPROVED** [review]

**Comment by sergiocorato** [comment]
@gurneyalex This PR is very old, but could be merged anyway I think

File: addons/stock_account/stock.py [hunk 1]
```diff
@@ -351,7 +351,8 @@ def _invoice_create_line(self, cr, uid, moves, journal_id, inv_type='out_invoice
                     invoice_line_vals['invoice_line_tax_id'] = extra_move_tax[0, move.product_id]
 
             move_obj._create_invoice_line_from_vals(cr, uid, move, invoice_line_vals, context=context)
-            move_obj.write(cr, uid, move.id, {'invoice_state': 'invoiced'}, context=context)
+        move_ids = [move.id for move in moves]
+        move_obj.write(cr, uid, move_ids, {'invoice_state': 'invoiced'}, context=context)
```
Comment (pedrobaeza) [inline addons/stock_account/stock.py:355]:
```suggestion
        move_obj.write(cr, uid, moves.ids, {'invoice_state': 'invoiced'}, context=context)
```

Comment (sergiocorato) [reply addons/stock_account/stock.py:355]: Done, thanks @pedrobaeza

Comment (sergiocorato) [reply addons/stock_account/stock.py:355]: This `moves` is not a recordset:
```
File "/home/travis/OCB-8.0/addons/stock_account/stock.py", line 354, in _invoice_create_line
move_obj.write(cr, uid, moves.ids, {'invoice_state': 'invoiced'}, context=context)
AttributeError: 'list' object has no attribute 'ids'
```
I restore previous code.

Comment (pedrobaeza) [reply addons/stock_account/stock.py:355]: Ah, OK

**Pull request closed without merging**